### PR TITLE
Add basic `unicorn` setup to ESLint config

### PR DESCRIPTION
**Pull request by codykaup**
This adds a very basic configuration for `unicorn`. I filtered a bunch of rules to prevent this PR from getting too big. The idea is to enable those rules over time one-by-one.

<details>
  <summary>📦 Published PR as canary version: <code>11.10.1--canary.1041.10835649865.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.10.1--canary.1041.10835649865.0
  # or 
  yarn add chromatic@11.10.1--canary.1041.10835649865.0
  ```
</details>

### CODE REVIEW

File: node-src/git/git.ts [hunk 1]
```diff
@@ -108,14 +108,13 @@ export async function getUncommittedHash() {
...
-      `(${listUncommittedFiles}) | git hash-object --stdin-paths | git hash-object --stdin`
-    )
-  ).trim();
+  let uncommittedHash = await execGitCommand(
+    // Pass the combined list of filenames to hash-object to retrieve a list of hashes. Then pass
+    // the list of hashes to hash-object again to retrieve a single hash of all hashes. We use
+    // stdin to avoid the limit on command line arguments.
+    `(${listUncommittedFiles}) | git hash-object --stdin-paths | git hash-object --stdin`
+  );
+  uncommittedHash = uncommittedHash.trim();
```
Comment (jmhobbs) [inline node-src/git/git.ts:117]: This one is interesting.  I kind of want the `const` here, no?

File: node-src/lib/logSerializers.test.ts [hunk 2]
```diff
@@ -7,8 +7,8 @@ it('strips off envPairs', () => {
   let err;
   try {
     execSync('some hot garbage', { stdio: 'ignore' });
-  } catch (e) {
-    err = e;
+  } catch (err_) {
+    err = err_;
```
Comment (jmhobbs) [inline node-src/lib/logSerializers.test.ts:11]:
```suggestion
  } catch (execError) {
    err = execError;
```

**Review by jmhobbs — APPROVED** [review]

File: node-src/git/git.ts (hunk 1 above)
Comment (codykaup) [reply node-src/git/git.ts:117]: I'm not against that! I figured just a whitespace method call wasn't a huge deal to use a `let`. ¯\\\_(ツ)\_\/¯ 

I'll get that updated!

File: node-src/lib/logSerializers.test.ts (hunk 2 above)
Comment (codykaup) [reply node-src/lib/logSerializers.test.ts]: Turns out, descriptive errors are already supported but it breaks when you add the `name` field for some reason? Once I removed that, I was getting the errors I expected.

**Comment by ghengeveld** [comment]
:rocket: PR was released in [`v11.10.1`](https://github.com/chromaui/chromatic-cli/releases/tag/v11.10.1) :rocket:

**Pull request merged**